### PR TITLE
=cls #18978 Lazy startup of shards when rememberEntities=false

### PR DESCRIPTION
**Pull request by patriknw**
- and don't populate the unallocatedShards Set in the State
  when rememberEntities=false

### CODE REVIEW

**Comment by patriknw** [comment]
Refs #18978

**Comment by patriknw** [comment]
Please review so that this can be included in 2.4.1 today.

File: akka-cluster-sharding/src/main/scala/akka/cluster/sharding/ShardCoordinator.scala [hunk 1]
```diff
@@ -809,7 +828,8 @@ class DDataShardCoordinator(typeName: String, settings: ClusterShardingSettings,
   // This state will drop all other messages since they will be retried
   def waitingForState: Receive = ({
     case g @ GetSuccess(CoordinatorStateKey, _) ⇒
-      state = g.get(CoordinatorStateKey).value
+      println(s"# GetSuccess $self ${settings.rememberEntities}") // FIXME
```
Comment (ktoso) [inline akka-cluster-sharding/src/main/scala/akka/cluster/sharding/ShardCoordinator.scala]: remove println please

Comment (patriknw) [inline akka-cluster-sharding/src/main/scala/akka/cluster/sharding/ShardCoordinator.scala]: oh, thanks

**Comment by ktoso** [comment]
LGTM, let's remove the println and merge

**Comment by drewhk** [comment]
LGTM

**Pull request merged**